### PR TITLE
Fix: Fix slow gcode parsing and parseValue bug

### PR DESCRIPTION
**Pull request by jane-rose**

### CODE REVIEW

**Comment by aligator** [comment]
Hi,
Thanks for your contribution :-)

Can you explain to me what's exactly wrong with the curve.getLength()? Is there a simple way I can reproduce the bug?

**Comment by jane-rose** [comment]
`curve.getLength()` takes too much time, same as creating a new array in `Array.reduce`

**Comment by aligator** [comment]
ok, sounds nice. I will do some benchmarking to verify this, but it should be ok.

Thank you :-)

File: src/parser.ts [hunk 1]
```diff
@@ -330,6 +332,7 @@ export class GCodeParser {
             } else if (cmd[0] === "G92") {
                 // set state
                 lastLastPoint.copy(lastPoint)
+                // TODO fix:  'parseValue' value may be zero which is also not truthy
```
Comment (jane-rose) [inline src/parser.ts:335]: Ops, one more

Comment (aligator) [reply src/parser.ts:335]: Nice find, 
should be fixed in https://github.com/aligator/gcode-viewer/pull/11
Could you please review that?

**Pull request merged**